docs/markdown/Dependencies.md

```diff
@@ -64,6 +64,26 @@ pkg-config files. Meson has autodetection support for some of these,
 and they are described [later in this
 page](#dependencies-with-custom-lookup-functionality).
 
+# Arbitrary variables from dependencies that can be found multiple ways
+
+*Note* new in 0.51.0
+
+When you need to get an arbarary variables from a dependency that can be
+found multiple ways, and you don't want to constraint the type you can use
+the generic `get_variable` method. This currently supports cmake, pkg-config,
+and config-tool based variables.
+
+```meson
+foo_dep = dependency('foo')
+var = foo.get_variable(cmake : 'CMAKE_VAR', pkgconfig : 'pkg-config-var', configtool : 'get-var', default : 'default')
+```
+
+It accepts the keywords 'cmake', 'pkgconfig', 'pkgconfig_define',
+'configtool', and 'default'. If the value cannot be found by the
+implementaiton specific method and default is defined then that value will be
+returned, otherwise an error raised. pkgconfig_define has the same meaning
+as 'define_variable' to `get_pkgconfig_variable`.
+
```

Comment:
This should probably also specify what happens when the variable name is not specified for the dependency type which was found (e.g. found by cmake, but cmake: kwarg is absent).

Reply:
I tried to resolve this, are you happy with the rewording?

Reply:
Makes sense